File: backend/src/main/java/com/team05/petmeeting/global/security/userdetails/CustomUserDetails.java
package com.team05.petmeeting.global.security.userdetails;

import java.util.Collection;
import java.util.List;
import lombok.RequiredArgsConstructor;
import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.core.userdetails.UserDetails;

@RequiredArgsConstructor
public class CustomUserDetails implements UserDetails {

    private final Long userId;
    private final List<GrantedAuthority> authorities;

    public Long getUserId() {
        return userId;
    }

    @Override
    public String getUsername() {
        return "";
    }

    @Override
    public Collection<? extends GrantedAuthority> getAuthorities() {
        return authorities;
    }

    @Override
    public String getPassword() {
        return "";
    }

    // 보안 정책 추가 구현 시 default boolean method Override
}

File: backend/src/main/java/com/team05/petmeeting/global/security/userdetails/CustomUserDetails.kt
package com.team05.petmeeting.global.security.userdetails

import org.springframework.security.core.GrantedAuthority
import org.springframework.security.core.userdetails.UserDetails

class CustomUserDetails(
    val userId: Long,
    private val authorities: List<GrantedAuthority>
) : UserDetails {

    override fun getUsername(): String = ""

    override fun getAuthorities(): Collection<GrantedAuthority> = authorities

    override fun getPassword(): String = ""

    // 필요 시 아래 기본 메서드들도 override 가능
    override fun isAccountNonExpired(): Boolean = true
    override fun isAccountNonLocked(): Boolean = true
    override fun isCredentialsNonExpired(): Boolean = true
    override fun isEnabled(): Boolean = true
}